### ISSUE TITLE
Removing Visible Products if Not Verified

### ISSUE THREAD

**Issue by dylviz** [avendure/VendurePlugin-seller-verification#5]
In the event that a user is Verified and has uploaded products, the superadmin may decide at some point to revoke the verification.

In this case, the existing products and product variants should be set to disabled/not-visible. This should remove them from being returned from any search query.

If the Seller becomes verified once again, then I believe it makes the most sense to NOT enable/show the products/product variants. The user will have to manually enable the products once again. This is because time may pass while they are disabled and we wouldn't want their products to start being purchased if they're no longer available

- [ ] Update tests to show that products are disabled from an existing Seller who has become unverified.

**Comment by ashishkpaul** [comment]
Yes, you can achieve this workflow with the SellerVerifyPlugin, but you’ll need to add some custom logic to handle product visibility when a seller’s verification status changes.

### **How to Implement This in SellerVerifyPlugin**

#### 1. **On Verification Revocation**
- When a seller is set to unverified:
  - Find all products and product variants belonging to that seller.
  - Set their `enabled` property to `false` (or equivalent field for visibility).
  - Optionally, log or notify the seller/admin of this change.

#### 2. **On Re-Verification**
- When a seller is re-verified:
  - **Do not** automatically re-enable their products.
  - The seller must manually re-enable each product/variant they want to sell again.

#### 3. **Effect on Search/Shop**
- Disabled products/variants are not returned in shop or search queries (this is default Vendure behavior).

---

### **Where to Add This Logic**

- In your `SellerVerifyService` (where you update the seller’s verification status), after changing the status:
  - If `isVerified` is set to `false`, run a query to disable all their products/variants.
  - If `isVerified` is set to `true`, do nothing to product visibility.

#### **Example Pseudocode**

```ts
if (!isVerified) {
  // Seller is being unverified
  const products = await productService.findBySellerId(sellerId);
  for (const product of products) {
    await productService.update(product.id, { enabled: false });
    // Optionally, also disable all variants
    for (const variant of product.variants) {
      await productVariantService.update(variant.id, { enabled: false });
    }
  }
}
// If re-verified, do nothing to product visibility
```

---

### **Summary of Workflow**

- **Unverify Seller:** All their products/variants are disabled (not visible in shop/search).
- **Re-verify Seller:** Products remain disabled. Seller must manually re-enable.
- **Default Vendure:** Disabled products/variants are not returned in shop/search queries.
